src/main/java/com/clouway/push/server/PushServiceImpl.java

```diff
@@ -36,16 +35,13 @@ public void pushEvent(PushEvent event) {
   @Override
   public void pushEvent(PushEvent event, String correlationId) {
 
-    // transforming the eventType
-    if (!Strings.isNullOrEmpty(correlationId)) {
-      event.getAssociatedType().setCorrelationId(correlationId);
-    }
+    String key = event.getKey() + correlationId;
 
     String message = eventSerializer.serialize(event);
```

Comment:
This will not work for events with correlationId - the subscription form the repository is correctly found and the event fired, but the message is serialized without the correlationId, and the client will not handle it, because the event key will not match.